pkg/segment/writer/stats/segstats.go

```diff
@@ -247,6 +222,13 @@ func AddSegStatsStr(segstats map[string]*SegStats, cname string, strVal string,
 
 		segstats[cname] = stats
 	}
+
+	floatVal, err := strconv.ParseFloat(strVal, 64)
+	if err == nil {
+		AddSegStatsNums(segstats, cname, SS_FLOAT64, 0, 0, floatVal, strVal, bb, aggColUsage, hasValuesFunc, hasListFunc)
+		return
+	}
+
 	stats.Count++
```

Comment:
Why do you increment `stats.Count` only if `strVal` can't be parsed as a float? Does it get incremented in `AddSegStatsNums`?

Reply:
The increment of Count will be handled in AddSegStatsNums. So no need to count twice.